pkg/controller/node/nodecontroller.go

```diff
@@ -312,6 +356,9 @@ func NewNodeController(
 						utilruntime.HandleError(fmt.Errorf("Error allocating CIDR: %v", err))
 					}
 				}
+				if nc.taintManager != nil {
```

Comment:
This is a bit strange. You are checking the same condition in two different ways:
1. "if ns.runtaintManager {"
2. "if nc.taintManager != nil {
 Can you please unify it?

Or even better, You can move all those checks to TaintManager implementation. E.g.
```
func (tm *taintManager) NodeUpdated(...) {
  if tm == nil {
    return
  }
  // Now real body
}
```

With this pattern you can omit all those "ifs" here and this code will be much cleaner and shorter.